citesphere/src/main/java/edu/asu/diging/citesphere/api/v1/user/AddNewItemController.java

```diff
@@ -0,0 +1,120 @@
+    @Autowired
+    private ICitationHelper citationHelper;
+
+    @Autowired
+    private ICitationManager citationManager;
+
+    @Autowired
+    private IUserManager userManager;
+
+    @Autowired
+    private IGilesUtil gilesUtil;
+
+    @Autowired
+    private IUploadFileJobManager jobManager;
+
+    @RequestMapping(value = "/groups/{groupId}/items/create", method = RequestMethod.POST, consumes = {
+        MediaType.MULTIPART_FORM_DATA_VALUE })
+    public ResponseEntity<Object> createNewItem(Principal principal,
+            @PathVariable("groupId") String zoteroGroupId, @ModelAttribute CitationForm itemWithGiles)
+            throws ZoteroConnectionException, GroupDoesNotExistException, ZoteroHttpStatusException,
+            ZoteroItemCreationFailedException {
+
+        IUser user = userManager.findByUsername(principal.getName());
+        if (user == null) {
+            return new ResponseEntity<>(HttpStatus.UNAUTHORIZED);
+        }
+        
+        if(itemWithGiles.getItemType() == null) {
+            logger.error("Missing Item Type");
+            return new ResponseEntity<>("Error: Missing Item Type", HttpStatus.BAD_REQUEST);
+        
+        }
+        ICitation citation = new Citation();
+        List<String> collectionIds = new ArrayList<>();
+        if (itemWithGiles.getCollectionId() != null && !itemWithGiles.getCollectionId().trim().isEmpty()) {
+            collectionIds.add(itemWithGiles.getCollectionId());
+        }
+        citation.setCollections(collectionIds);
+        citationHelper.updateCitation(citation, itemWithGiles, user);
+
+        try {
+            citation = citationManager.createCitation(user, zoteroGroupId, collectionIds, citation);
+        } catch (ZoteroItemCreationFailedException e) {
+            return new ResponseEntity<>("Error: Zetero Item creation failed. " + e.getLocalizedMessage(), HttpStatus.INTERNAL_SERVER_ERROR);
+        }
+
+        if (itemWithGiles.getFiles() != null && itemWithGiles.getFiles().length > 0) {
+            ObjectMapper mapper = new ObjectMapper();
+            ObjectNode root = mapper.createObjectNode(); 
+            for (int i=0; i<itemWithGiles.getFiles().length; i++) {
+                IGilesUpload job;
+                try {
+                    job = jobManager.createGilesJob(user, itemWithGiles.getFiles()[i], itemWithGiles.getFiles()[i].getBytes(), zoteroGroupId,
+                            citation.getKey());
+                } catch (GroupDoesNotExistException e) {
+                    logger.error("Could not create job because group does not exist.", e);
+                    return new ResponseEntity<>("Error: Could not create job because group does not exist.", HttpStatus.BAD_REQUEST);
+                } catch (CannotFindCitationException e) {
+                    logger.error("Could not find the newly created citation.", e);
+                    return new ResponseEntity<>("Error: Could not find the newly created citation.", HttpStatus.BAD_REQUEST);
```

Comment:
I think in this case, we want to return a 500. Afterall, there is nothing the user could do here, it would indicate some internal issue.